xds/server.go

```diff
@@ -145,6 +146,12 @@ func (s *GRPCServer) RegisterService(sd *grpc.ServiceDesc, ss interface{}) {
 	s.gs.RegisterService(sd, ss)
 }
 
+// GetServiceInfo returns a map from service names to ServiceInfo.
+// Service names include the package names, in the form of <package>.<service>.
+func (s *GRPCServer) GetServiceInfo() map[string]grpc.ServiceInfo {
+	return s.gs.GetServiceInfo()
```

Comment:
@easwars We discussed embedding `gs`, do you remember why we decided to not do it eventually?

Reply:
I don't remember exactly why we decided not to embed it. But I can see multiple reasons for it:
- there are methods on `grpc.Server` which we don't want on `xds.GRPCServer` like `ServeHTTP`
- we want the `GRPCServer.gs` field to be an interface instead of a concrete type to able to use a fake `grpc.Server` in some tests

Reply:
Hmm .. actually we could embed an interface as well, not just a concrete type. So, that argument does not hold good. But the first point still makes sense.

Reply:
The only two methods can inherit are `Register` and this new one. The others all need override to do something.
So the loss of readability probably doesn't justify the gain.